### PR TITLE
Case 22092: Fix avatar/shape material targets, refresh material target when parent changes (RC82)

### PR DESCRIPTION
**Pull request by dback2**
https://highfidelity.fogbugz.com/f/cases/22092/

Master PR https://github.com/highfidelity/hifi/pull/15352

### CODE REVIEW

File: scripts/system/edit.js [hunk 1]
```diff
@@ -2523,18 +2523,24 @@ var PropertiesTool = function (opts) {
...
-				materialTargetData: parentModelData,
-			});
-		}
+            var parentModelData;
+            var properties = Entities.getEntityProperties(data.entityID, ["type", "parentID"]);
+            if (properties.type === "Material" && properties.parentID !== Uuid.NULL) {
+                var parentType = Entities.getEntityProperties(properties.parentID, ["type"]).type;
+                if (parentType === "Model" || Entities.getNestableType(properties.parentID) === "avatar") {
+                    parentModelData = Graphics.getModel(properties.parentID);
+                }
```
Comment (thoys) [inline scripts/system/edit.js]: style: move collapse `else if (` together with the `}` on the previous line

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18646/HighFidelity-Beta-PR15351-053669f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18646/HighFidelity-Beta-Interface-PR15351-053669f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18646/HighFidelity-Beta-PR15351-053669f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18646/HighFidelity-Beta-Interface-PR15351-053669f.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18646/HighFidelity-Beta-PR15351-053669f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18646/HighFidelity-Quest-Beta-PR15351-053669f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18689/HighFidelity-Beta-PR15351-f94c78e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18689/HighFidelity-Beta-Interface-PR15351-f94c78e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18689/HighFidelity-Beta-PR15351-f94c78e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18689/HighFidelity-Quest-Beta-PR15351-f94c78e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18689/HighFidelity-Beta-PR15351-f94c78e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18689/HighFidelity-Beta-Interface-PR15351-f94c78e.exe

**Pull request merged**